caddy: update to 2.6.1.

TinfoilSubmarine opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: **YES**

<!--
#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**|**NO**
-->

<!-- Note: If the build is likely to take more than 2 hours, please add ci skip tag as described in
https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->
<!--
#### Local build testing
- I built this PR locally for my native architecture, (ARCH-LIBC)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl
  - armv7l
  - armv6l-musl
-->

steinex commented:
Even if the patch does not apply anymore it is still needed and needs to be adapted.

Other than that I want to propose to remove the INSTALL file, we're on Caddy 2 for a very long time now.

TinfoilSubmarine commented:
> Even if the patch does not apply anymore it is still needed and needs to be adapted.

Thanks, fixed.

> Other than that I want to propose to remove the INSTALL.msg file, we're on Caddy 2 for a very long time now.

```
commit f9013e2b7a503570804b60b6cbdea967779b3fb7
Author: Patrick Pichler <git@patrickpichler.dev>
Date:   Thu Apr 15 15:44:37 2021 +0200

    caddy: upgrade to 2.4.3
    
    As there is currently a bug with go build info a patch is needed to
    fix printing the correct version number of the binary. Once the bug is
    fixed, the patch can be dropped.
    
    Fixes #22092
```

I agree, removed.
